process/process_test.go

```diff
@@ -1481,6 +1481,126 @@ var _ = Describe("Process", func() {
 				Expect(quick.Check(loop, nil)).To(Succeed())
 			})
 		})
+
+		Context("when we are in the proposing step", func() {
+			Context("and step into prevoting step", func() {
+				Context("with >= 2f+1 prevotes", func() {
+					It("should schedule a prevote timeout", func() {
+						loop := func() bool {
+							currentHeight := process.Height(r.Int63())
+							currentRound := process.Round(r.Int63())
+							whoami := id.NewPrivKey().Signatory()
+							f := 5 + (r.Int() % 10)
+							timerOptions := timer.
+								DefaultOptions().
+								WithTimeout(1 * time.Millisecond).
+								WithTimeoutScaling(0)
+							timeoutSignal := make(chan timer.Timeout, 2)
+							handlePrevoteTimeout := func(timeout timer.Timeout) {
+								timeoutSignal <- timeout
+							}
+							timer := timer.NewLinearTimer(timerOptions, nil, handlePrevoteTimeout, nil)
+
+							p := process.New(whoami, f, timer, nil, nil, nil, nil, nil, nil)
+
+							p.StartRound(currentRound)
+							p.CurrentHeight = currentHeight
+
+							// feed >= 2f+1 prevotes
+							t := 2*f + 1 + r.Intn(10)
+							for i := 0; i < t; i++ {
+								prevote := randomValidPrevoteMsg(r, id.NewPrivKey().Signatory(), currentHeight, currentRound)
+								p.Prevote(prevote)
+							}
+
+							// make sure we haven't received any prevote timeout
+							select {
+							case _ = <-timeoutSignal:
+								Expect(true).To(BeFalse())
+							default:
+								Expect(true).To(BeTrue())
+							}
```

Comment:
Generally nicer to explicitly state the error message by doing something like:

```suggestion
							select {
							case _ = <-timeoutSignal:
								Fail("received a prevote timeout")
							default:
							}
```